### ISSUE TITLE
exp: failed runs with `--temp` go missing

### ISSUE THREAD

**Issue by gregstarr** [iterative/dvc#10616]
# Bug Report

## Description

experiments run with `--temp` that fail appear to go missing. Sounds like this was an issue previously but was fixed, however I still am having this problem. I am using a recent version of DVC and the previous issue was from 2 years ago.

https://github.com/iterative/dvc/issues/8612

### Reproduce

1. `dvc exp run --temp`
2. `dvc exp show` properly lists experiment
3. first stage success
4. second stage fail
5. `dvc exp show` doesn't show the failed experiment

### Expected
1.  `dvc exp show` shows the failed experiment
2. if for some reason it shouldn't show the failed experiment, then note in the docs because this seems like a fairly significant behavior difference between running with the queue and with temp

### Environment information

**Output of `dvc doctor`:**
```bash
DVC version: 3.55.2 (pip)
-------------------------
Platform: Python 3.11.10 on Linux-3.10.0-693.el7.x86_64-x86_64-with-glibc2.17
Subprojects:
        dvc_data = 3.16.6
        dvc_objects = 5.1.0
        dvc_render = 1.0.2
        dvc_task = 0.4.0
        scmrepo = 3.3.8
Supports:
        http (aiohttp = 3.10.8, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.10.8, aiohttp-retry = 2.8.3)
Config:
        Global: /home/starrgw1/.config/dvc
        System: /etc/xdg/dvc
```

**Comment by gregstarr** [comment]
Here is the experiment that failed that I can't find, this is the JSON from the tmp/exps/run directory:

```json
{
  "git_url": "pathpathpath/.dvc/tmp/exps/standalone/tmpg4hu6_2a",
  "baseline_rev": "27e681c74670e183f4bdcbf4aa9434e8994f2c19",
  "location": "tempdir",
  "root_dir": "pathpathpath/.dvc/tmp/exps/standalone/tmpg4hu6_2a",
  "dvc_dir": ".dvc",
  "name": "depth_normalization_3",
  "wdir": ".",
  "result_hash": null,
  "result_ref": null,
  "result_force": false,
  "status": 1
}
```

**Comment by gregstarr** [comment]
Wow I can `git cat-file` the sha from the name of the json file:

```bash
> tree a55df7458541c3aeecc816840830de454d66472d
parent ce7991feda5aabbfa18d9541c5d90191850b7b09
parent 1e3c6c8e652abbaf9fd1f47a63cdeadb4e8b5697
author starrgw1 email 1730829745 +0000
committer starrgw1 email 1730829745 +0000

dvc-exp:ce7991feda5aabbfa18d9541c5d90191850b7b09:ce7991feda5aabbfa18d9541c5d90191850b7b09:mean_depth_norm
```

And in fact I can apply the commit through either git or DVC. However, it doesn't look like I can get back the stage 1 results? `dvc.lock` seems unchanged, so nothing gets checkout out.

**Comment by shcheklein** [comment]
@gregstarr could you share a script / project to reproduce this?

**Comment by gregstarr** [comment]
Here is a minimal example: https://github.com/gregstarr/minimum-dvc

1. clone
2. install a recent version of dvc
3. `dvc exp run --temp --name "exp1"`
4. `dvc exp show -A`

the third command should fail in the third stage, then the fourth command should't show the experiment "exp1". I would expect `dvc exp show -A` to show the failed experiment.

**Comment by pmrowla** [comment]
IIRC this is expected behavior for `--temp` runs. Failed experiments are only collected when run via the queue. The failure state for a queued experiment run is tied to the celery/dvc-task job for that run, but `--temp` runs don't touch celery at all.

In the codebase you can see that queue runs have an implementation for `collect_failed_data`
https://github.com/iterative/dvc/blob/1e08cc5da3ded5ca6d67fc1941ceb46181d1ec8b/dvc/repo/experiments/queue/celery.py#L569

but workspace runs do not (and `--temp` runs extend the workspace implementation)
https://github.com/iterative/dvc/blob/1e08cc5da3ded5ca6d67fc1941ceb46181d1ec8b/dvc/repo/experiments/queue/workspace.py#L250-L255

**Comment by shcheklein** [comment]
@pmrowla yep, thanks. But what was the reason for this? (if you remember :) )

**Comment by pmrowla** [comment]
It's leftover from `--temp` predating the queue. Workspace runs have never had any kind of saved execution state (other than the git commit on success), and `--temp` was originally just the extension of "do exactly what we do in a workspace run except in an isolated directory".

**Comment by gregstarr** [comment]
I have a shell script which calls `dvc exp run --temp`, which is nice because the command blocks until the pipeline completes. How would I replicate this behavior using the queue so that I can see when experiments fail and examine the failed state?

**Comment by shcheklein** [comment]
@gregstarr how about analyzing the results of the `dvc queue status` command?

(I wonder if we should just do `dvc queue wait` or something)

**Comment by gregstarr** [comment]
so have python or bash call `dvc queue status` periodically and check to see if the queue is empty?

**Comment by shcheklein** [comment]
yep, something like that
